### ISSUE TITLE
networkd: add a way to Match no device

### ISSUE THREAD

**Issue by nolange** [systemd/systemd#14631]
**Is your feature request related to a problem? Please describe.**
We have some ideally read-only rootfs with pre-configured network, but dont know the names of the devices until boot.
The idea is to have anything but the matching devicename in the rootfs, and during boot-up (pre-network) figure out the correct device.

**Describe the solution you'd like**

Well, I already have a kinda working solution, but its rather unintuitive, and your manual is rather unspecific about what exactly happens here.
I guess the final test if if Name is either `X` **or** `enp4s0`, despite the manual reading like it should test for `X` **and** `enp4s0`.
Even weirder, I tried with `Name=!*` in the `etc` unit, and this will not match, so here it appears to test for `dont match everything` **and** `match enp4s0`.

In short, Id like to get some pointers how I could/should do this correctly, and possibly if some new key could help inverting the default of "Match all" 

_/etc/systemd/network/20-ccpu.network_

```ini
[Match]
# Doesnt match anything (hopefully)
Name=X

[Network]
Description="C-CPU Network"
Address=192.168.8.1/30
NTP=192.168.8.2
```
_/run/systemd/network/20-ccpu.network.d/matchname.conf_

```ini
[Match]
Name=enp4s0
```
